service/storage_service.cc

```diff
@@ -253,15 +253,15 @@ static future<> set_gossip_tokens(gms::gossiper& g,
  */
 future<> storage_service::wait_for_ring_to_settle(std::chrono::milliseconds delay) {
     // first sleep the delay to make sure we see *at least* one other node
-    for (int i = 0; i < delay.count() && _gossiper.get_live_members().size() < 2; i += 1000) {
-        co_await sleep_abortable(std::chrono::seconds(1), _abort_source);
+    for (int i = 0; i < delay.count() && _gossiper.get_live_members().size() < 2; i += 10) {
+        co_await sleep_abortable(std::chrono::milliseconds(10), _abort_source);
     }
 
     auto t = gms::gossiper::clk::now();
     while (true) {
+        slogger.info("waiting for schema information to complete");
```

Comment:
You could also do the prints inside the loop but use `logger::rate_limit` to limit them to e.g. once every 1 seconds as it was, or even less frequently.

Reply:
Maybe, but what's the point in printing this message periodically instead of just once when starting to wait?

Reply:
No point.
For a while I was worried that it decreases debuggability (we don't see whether we're stuck on waiting for schema or we got past that point) - but there's `slogger.info("Checking bootstrapping/leaving nodes: ok")` at the end of the function so we can see where we are.